ml-agents-envs/mlagents/envs/subprocess_env_manager.py

```diff
@@ -0,0 +1,179 @@
+from typing import *
+import cloudpickle
+
+from mlagents.envs import UnityEnvironment
+from multiprocessing import Process, Pipe
+from multiprocessing.connection import Connection
+from mlagents.envs.base_unity_environment import BaseUnityEnvironment
+from mlagents.envs.env_manager import EnvManager, StepInfo
+from mlagents.envs import AllBrainInfo, BrainParameters, ActionInfo
+
+
+class EnvironmentCommand(NamedTuple):
+    name: str
+    payload: Any = None
+
+
+class EnvironmentResponse(NamedTuple):
+    name: str
+    worker_id: int
+    payload: Any
+
+
+class UnityEnvWorker(NamedTuple):
+    process: Process
+    worker_id: int
+    conn: Connection
+
+    def send(self, name: str, payload=None):
+        try:
+            cmd = EnvironmentCommand(name, payload)
+            self.conn.send(cmd)
+        except (BrokenPipeError, EOFError):
+            raise KeyboardInterrupt
```

Comment:
I understand this is probably reproducing the old behavior, but converting these to a KeyboardInterrupt feels strange. Maybe convert all three to a new EnvironmentDoneException (or something like that)?

Reply:
I agree (and should have done it to begin with) -- but this PR has grown a bit large.  Mind if I submit this as a subsequent PR?

Reply:
Sounds good